### PR TITLE
[lldb] Unlock Swift scratch context before early return

### PR DESCRIPTION
**Pull request by kastiglione**
rdar://85399160

### CODE REVIEW

**Comment by kastiglione** [comment]
@swift-ci test

**Comment by kastiglione** [comment]
This `return` is within a successful `try_lock` block, but does not unlock the lock before returning (`GetSwiftScratchContextLock`). Here's the relevant code https://github.com/apple/llvm-project/blob/cb6c718b794584f3e8bd6a2ccc3592f8ce22c5d6/lldb/source/Target/Target.cpp#L2284-L2288

It seems this could cause deadlock on a second call to `GetScratchTypeSystemForLanguage`.

File: lldb/source/Target/Target.cpp [hunk 1]
```diff
@@ -2309,8 +2309,10 @@ Target::GetScratchTypeSystemForLanguage(lldb::LanguageType language,
           m_scratch_type_system_map.RemoveTypeSystemsForLanguage(language);
           type_system_or_err = m_scratch_type_system_map.GetTypeSystemForLanguage(
               language, this, create_on_demand, compiler_options);
-          if (!type_system_or_err)
```
Comment (adrian-prantl) [inline lldb/source/Target/Target.cpp]: This definitely looks correct. Maybe we should use `llvm::scope_exit([&](){GetSwiftScratchContextLock().unlock();});` right at the beginning to cover all paths.

Comment (kastiglione) [reply lldb/source/Target/Target.cpp]: done

**Comment by kastiglione** [comment]
@swift-ci test

**Review by augusto2112 — APPROVED** [review]

**Pull request merged**